src/pd/util.rs

```diff
@@ -389,16 +388,17 @@ pub fn try_connect_leader(
     env: Arc<Environment>,
     previous: &GetMembersResponse,
 ) -> Result<(PdClient, GetMembersResponse)> {
-    // Try to connect other members.
-    // Randomize endpoints.
+    let previous_leader = previous.get_leader();
     let members = previous.get_members();
-    let mut indexes: Vec<usize> = (0..members.len()).collect();
-    rand::thread_rng().shuffle(&mut indexes);
-
     let cluster_id = previous.get_header().get_cluster_id();
     let mut resp = None;
-    'outer: for i in indexes {
-        for ep in members[i].get_client_urls() {
+    // Try to connect to other members, then the previous leader.
+    'outer: for m in members
+        .into_iter()
+        .filter(|m| *m == previous_leader)
```

Comment:
`==` should be `!=`?

Reply:
the filter will skip the previous leader, maybe fine here.